mage/query-modules/cpp/periodic.md

```diff
@@ -0,0 +1,137 @@
+---
+id: periodic
+title: periodic
+sidebar_label: periodic
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import RunOnSubgraph from '../../templates/_run_on_subgraph.mdx';
+
+export const Highlight = ({children, color}) => (
+<span
+style={{
+  backgroundColor: color,
+  borderRadius: '2px',
+  color: '#fff',
+  padding: '0.2rem',
+}}>
+{children}
+</span>
+);
+
+**Periodic module** enables users to execute a query periodically in batches. Periodic in this case
+does not mean that the query is executed after a specific timeout. It actually signifies that the results
+of some input source is being batched in order to speed up the execution time, due to the complexity of the
+query. 
+
+[![docs-source](https://img.shields.io/badge/source-graph_util-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/cpp/graph_util_module)
+
+| Trait               | Value                                                                                                     |
+| ------------------- | --------------------------------------------------------------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**module**</Highlight>                                                         |
+| **Implementation**  | <Highlight color="#FB6E00">**C++**</Highlight>                                                           |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>                                                   |
+
+### Procedures
+
+### `iterate(input_query, running_query, params)`
+
+:::caution
+This module breaks the isolation levels in the database, as the results are batched and executed in different transactions. Do
+not consider the query which executes this procedure to be aligned with set isolation level of the database, as every executed batch
+is committed by itself. An issue while running this procedure will not result in rollbacking batches of committed results.
+:::
+
+
+
+#### Input:
+
+- `input_query: string` ➡ the input query which will yield the results that need to be batched
+- `running_query: string` ➡ query which will be executed on the batched results
+- `params: Map[string, string]` ➡ parameters for the procedure
```

Comment:
there will be more of them in the future?

Reply:
not sure

Reply:
possibly, but currently no need